### PR TITLE
Feature/mesh offset thicken 2

### PR DESCRIPTION
**Pull request by robin-oval**
new mesh_offset and mesh_thicken functions
I do not know why
from compas.datastructures.mesh import mesh_flip_cycles
does not work

### CODE REVIEW

**Comment by tomvanmele** [comment]
there seems to be quite a bit of overlap with one of your other pull requests. can we split this properly over two and merge them in the right order?

**Comment by tomvanmele** [comment]
also, it is a bit unclear what you mean by `mesh_flip_cycles` does not work, because i just ran it on a few examples and it seems to work just fine...

**Comment by robin-oval** [comment]
sometimes the second level import does not work in a new branch but works in the master branch...

File: src/compas/datastructures/mesh/join.py [hunk 1]
```diff
@@ -4,57 +4,105 @@
 
 from copy import deepcopy
 
-from compas.utilities import geometric_key
-
-from compas.datastructures.mesh.clean import mesh_delete_duplicate_vertices
+from compas.utilities import pairwise
```
Comment (gonzalocasas) [inline src/compas/datastructures/mesh/join.py]: this doesn't seem to be used.

**Comment by tomvanmele** [comment]
@robin-oval 
> sometimes the second level import does not work in a new branch but works in the master branch...

i was commenting on a different case

```python
# in compas.topology.xxx.py
from compas.datastructures.network import Network
# should be
from compas.datastructures import Network
```
if you are importing from within the same package you should import from the same level as the module you are importing into...
```python
# in compas.datastructures.mesh.geodesics.py
from compas.datastructures import trimesh_cotangent_laplacian_matrix
# should be
from compas.datastructures.mesh.matrices import trimesh_cotangent_laplacian_matrix
```

**Comment by robin-oval** [comment]
actually this PR calls a function included in another PR

should I remove the older PR or wait for it to be merged?

**Pull request merged**